feat(trace_to_log transform): introduces transform that converts traces to logs (#24168)
vectordotdev/vector: changelog.d/add-trace-to-log-transform.feature.md

> Adds transform that allows converting traces to logs. It is very simple and piggy backs on the fact that traces are thin wrappers around logs. 

Reviewer: ```suggestion
Introduced `trace_to_log` transform that allows converting traces to logs.
```
